p2p/enode/urlv4.go

```diff
@@ -151,6 +151,19 @@ func parseComplete(rawurl string) (*Node, error) {
 			return nil, errors.New("invalid discport in query")
 		}
 	}
+	// Check if hostname is an IP address and create node accordingly
+	hostname := u.Hostname()
+	ip := net.ParseIP(hostname)
+	if ip == nil {
+		ips, err := lookupIPFunc(hostname)
```

Comment:
I might be wrong here, but if I understand correctly this will not create a correct V4WithDNS node if the initial dial is working. Shouldn't we in this case always create NewV4WithDNS even if the DNS lookup is working (so we can get the DNS refresh ability). Otherwise we will kinda have three classes of nodes: raw-ip, DNS resolved on startup, DNS not resolved on startup all with slightly different behavior